src/Storages/MergeTree/MergeTreeIndexFullText.cpp

```diff
@@ -548,12 +548,21 @@ bool MergeTreeConditionFullText::traverseTreeEquals(
         token_extractor->stringLikeToBloomFilter(value.data(), value.size(), *out.bloom_filter);
         return true;
     }
-    else if (function_name == "hasToken")
+    else if (function_name.starts_with("hasToken"))
     {
         out.key_column = key_column_num;
         out.function = RPNElement::FUNCTION_EQUALS;
         out.bloom_filter = std::make_unique<BloomFilter>(params);
-        const auto & value = const_value.get<String>();
+        const auto value = std::invoke(
```

Comment:
Is the `std::invoke` machinery really needed or can't l. 559-561 be called directly?

Reply:
Well.. in cases such as this I use `std::invoke` to still be able to enforce const correctness on variables that are initialized in a more complex manner than a single expression would allow for. I do agree that it hurts readability, so I'll just get rid of it in this instance because the enclosing block is rather short.